indico/htdocs/js/indico/modules/registration/reglists.js

```diff
@@ -184,8 +184,18 @@
         });
     }
 
+
     global.setupRegistrationListFilter = function setupRegistrationListFilter() {
+        $(document).on('click', function(){
```

Comment:
Bind it to something inside the dialog window. The dialog is part of the normal document so like this you bind it again and again everytime you open the filter dialog.